cvmfs/sync_item.h

```diff
@@ -155,15 +198,18 @@ class SyncItem {
   std::string GetGraftMarkerPath() const;
   void CheckGraft();
 
-  const SyncUnion *union_engine_;
+  const SyncUnion *union_engine_;     /**< this SyncUnion created this object */
 
   mutable EntryStat rdonly_stat_;
   mutable EntryStat union_stat_;
   mutable EntryStat scratch_stat_;
 
-  bool whiteout_;
-  bool valid_graft_;
-  bool graft_marker_present_;
+  bool whiteout_;                     /**< SyncUnion marked this as whiteout  */
+  bool opaque_;                       /**< SyncUnion marked this as opaque dir*/
+  bool masked_hardlink_;              /**< SyncUnion masked out the linkcount */
+  bool valid_graft_;                  /**< ??? */
+  bool graft_marker_present_;         /**< ??? */
```

Comment:
This might ask for some documentation strings. I'm not sure what exactly those members do, however.